test/api/model/model_checks.dart

```diff
@@ -23,21 +15,9 @@ extension ReactionsChecks on Subject<List<Reaction>> {
   void deepEquals(_) {
     throw UnimplementedError('Tried to call [Subject<List<Reaction>>.deepEquals]. Use jsonEquals instead.');
   }
```

Comment:
Is it intentional to leave in this `ReactionsChecks on Subject<List<Reaction>>`, with its `deepEquals` implementation that throws `UnimplementedError`?

The error still does apply, I think. It's `jsonEquals`, not `deepEquals`, that actually looks deeply enough to be useful in a test, given that `Reaction` doesn't have an `==` override. So the error is likely to be helpful when you encounter it.

But if the error is needed, then probably the need won't be unique to `List<Reaction>`s, right? Does it look odd to you to have this narrow extension on `Subject<List<Reaction>>`?

Reply:
Yeah, this would be equally appropriate on `List` of any type that we know doesn't have a `==` that you would want in a test, and that has a `toJson` that you'd want instead.

It does look a bit odd having it on this one type. But it was already written, and could be useful, so I didn't feel like deleting it. I probably wouldn't go around trying to systematically add these for other types that could equally use them.